### PR TITLE
Add documentation for Holiday integration

### PR DESCRIPTION
**Pull request by jrieger**
## Proposed change

Add documentation for Holiday integration

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/103795
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/4888
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by jpbede — DISMISSED** [review]
It looks like you created your branch from `current`, could you please rebase your branch from `next` as it contains unrelated commits.

File: source/_integrations/holiday.markdown [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ha_integration_type: integration
+---
+
+The Holiday integration allows you to create a calendar of holidays in Home Assistant for powering automations.
+
+A calendar entity has a state and attributes that represent the next upcoming event (only). A calendar trigger is a much more flexible way to power automations with fewer limitations than using the entity state.
+
+{% include integrations/config_flow.md %}
+
+## Calendar Event Automations
```
Comment (gjohansson-ST) [inline source/_integrations/holiday.markdown]: I suppose this section should/could give proposals on holiday calendar and not calendar automation in general (that could be a reference below)

File: source/_integrations/holiday.markdown [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ha_config_flow: true
+ha_domain: holiday
+ha_platforms:
+  - calendar
+ha_codeowners:
+  - '@jrieger'
+ha_integration_type: integration
+---
+
+The Holiday integration allows you to create a calendar of holidays in Home Assistant for powering automations.
```
Comment (gjohansson-ST) [inline source/_integrations/holiday.markdown:17]: Probably we should ref the upstream library like we do in [holiday](https://www.home-assistant.io/integrations/workday/)

File: source/_integrations/holiday.markdown [hunk 3]
```diff
@@ -0,0 +1,31 @@
+---
+title: Holiday
+description: Instructions on how to use holiday calendars in Home Assistant.
+ha_category:
+  - Calendar
+ha_iot_class: Local Polling
+ha_release: 2023.12
```
Comment (gjohansson-ST) [inline source/_integrations/holiday.markdown]:
```suggestion
ha_release: 2024.1
```

**Review by gjohansson-ST — DISMISSED** [review]

File: source/_integrations/holiday.markdown (hunk 1 above)
Comment (jrieger) [reply source/_integrations/holiday.markdown]: Done.

File: source/_integrations/holiday.markdown (hunk 2 above)
Comment (jrieger) [reply source/_integrations/holiday.markdown:17]: Done.

File: source/_integrations/holiday.markdown [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+{% include integrations/config_flow.md %}
+
+## Holiday calendar automations
+
+Individual holiday *Events* are what powers automations such as:
+
+- Turn on the Christmas lights at the *start* of Christmas Day
+- Prevent your covers from being opened on public holidays
+
+See [Calendar Automations](/integrations/calendar#automation) for an overview, and read more about [Calendar Trigger Variables](/docs/automation/templating/#calendar) for the available information you can use in a condition or action such as the event `summary`, `description`, `location` and more.
```
Comment (c0ffeeca7) [inline source/_integrations/holiday.markdown]:
```suggestion
- Turn on the Christmas lights at the *start* of Christmas Day.
- Prevent your covers from being opened on public holidays.

For an overview, refer to [calendar automations](/integrations/calendar#automation). You can use [calendar trigger variables](/docs/automation/templating/#calendar) in a condition or action such as the event `summary`, `description`, `location` and in other places.
```

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @jrieger and @gjohansson-ST :+1:

**Pull request merged**